scripts/createSingleNav.js

```diff
@@ -0,0 +1,47 @@
+const fs = require('fs');
+const path = require('path');
+const yaml = require('js-yaml');
+
+const createSingleNav = () => {
+  const rootNavYaml = yaml.safeLoad(
+    fs.readFileSync(path.join(process.cwd(), 'src/nav/root.yml'), 'utf8')
+  );
+
+  const ignorePages = [
+    '/docs/glossary/glossary',
+    '/attribute-dictionary',
+    '/whats-new',
+    '/docs/release-notes',
+  ];
+
+  const rootNavChildren = rootNavYaml.pages.reduce((acc, page) => {
+    let subNavChildren = null;
+    if (
+      page.path &&
+      !page.path.startsWith('http') &&
+      !ignorePages.includes(page.path)
+    ) {
+      const pageDirectory = page.path.split('/')[2];
+      subNavChildren = yaml.safeLoad(
+        fs.readFileSync(
+          path.join(process.cwd(), `src/nav/${pageDirectory}.yml`),
+          'utf8'
+        )
+      );
+    }
+    if (subNavChildren) {
+      acc.push({ ...page, pages: [subNavChildren] });
+    } else {
+      acc.push({ ...page });
+    }
+    return acc;
```

Comment:
[nit] someone once told me when i used `reduce()` you could also write it without side effect of `.push()` if you're so inclined. something like:

```
return subNavChildren ? {...acc, { ...page, pages: [subNavChildren]} } : {...acc, { ...page }}
```